engines/scumm/smush/codec1.cpp

```diff
@@ -22,40 +22,18 @@
 
 #include "common/endian.h"
 
+#include "scumm/bomp.h"
+
 namespace Scumm {
 
 void smushDecodeRLE(byte *dst, const byte *src, int left, int top, int width, int height, int pitch) {
-	byte val, code;
-	int32 length;
-	int h = height, lineSize;
-
 	dst += top * pitch;
-	for (h = 0; h < height; h++) {
-		lineSize = READ_LE_UINT16(src);
-		src += 2;
+	do {
 		dst += left;
-		while (lineSize > 0) {
-			code = *src++;
-			lineSize--;
-			length = (code >> 1) + 1;
-			if (code & 1) {
-				val = *src++;
-				lineSize--;
-				if (val)
-					memset(dst, val, length);
-				dst += length;
-			} else {
-				lineSize -= length;
-				while (length--) {
-					val = *src++;
-					if (val)
-						*dst = val;
-					dst++;
-				}
-			}
-		}
-		dst += pitch - left - width;
```

Comment:
Well, I haven't seen references of `- width` in the disasm... but then again, my knowledge of that part is very modest. The whole line (with `pitch` and `left`) was added by sev about 17 years ago. I guess it had a purpose back then. But now I really can't tell, and I couldn't see any differences playing FT for a while on the current master build and your build...